BlockChain: use Network in constructors

msgilligan opened the pull request:
Constructors that take `NetworkParameters` are marked as `@VisibleForTesting`. This is because of the special `UNITTEST` `NetworkParameters` that is required for certain tests.

schildbach commented on core/src/main/java/org/bitcoinj/core/AbstractBlockChain.java:
```diff
@@ -160,13 +162,26 @@ static class OrphanBlock {
 
+    /**
+     * Constructs a BlockChain connected to the given list of listeners (wallets) and a store.
+     * @param network network parameters for this chain
```
Is "parameters" accurate?

msgilligan replied:
Fixed and force-pushed.

schildbach reviewed (commented):
Looks good, except for 1 minor comment.

msgilligan commented:
> Looks good, except for 1 minor comment.

I fixed the comment text and force-pushed. I also removed two of the `@VisibleForTesting` constructors in `Blockchain` because they weren't being used and didn't seem necessary.
